src/native/http2_stream_manager.c

```diff
@@ -312,10 +312,10 @@ static void s_on_stream_acquired(struct aws_http_stream *stream, int error_code,
                 env, callback_data->java_async_callback, async_callback_properties.on_failure, crt_exception);
             (*env)->DeleteLocalRef(env, crt_exception);
             (*env)->ExceptionClear(env);
-            aws_http_stream_binding_destroy(env, callback_data->stream_binding);
+            aws_http_stream_binding_release(env, callback_data->stream_binding);
         } else {
-            /* Stream is activated once we acquired from the Stream Manager */
-            aws_atomic_store_int(&callback_data->stream_binding->activated, 1);
+            /* Acquire for the native stream */
+            aws_http_stream_binding_acquire(callback_data->stream_binding);
```

Comment:
don't we need to do binding_acquire() as long as `native_stream` exists? because the destroy_callback will decref the binding?